src/Types/GroupMetadata.ts

```diff
@@ -4,6 +4,10 @@ export type GroupParticipant = (Contact & { isAdmin?: boolean, isSuperAdmin?: bo
 
 export type ParticipantAction = 'add' | 'remove' | 'promote' | 'demote'
 
+export type RequestJoinAction = 'created' | 'revoked'
+
+export type RequestJoinMethod = 'invite_link' | 'linked_group_join' | 'non_admin_add' | undefined
```

Comment:
can you describe when each of these join methods appear?

Reply:
invite_link = from a link
linked_group_join = from a community
non_admin_add = a non admin member add